frontend/ui/widget/datetimewidget.lua

```diff
@@ -101,57 +137,148 @@ function DateTimeWidget:init()
+            value_min = self.month_min or 1,
+            value_max = self.month_max or 12,
+            value_step = 1,
+            value_hold_step = self.month_hold_step or 3,
+            width = number_picker_widgets_width,
+        }
+        self:mergeLayoutInHorizontal(month_widget)
+    else
+        month_widget = dummy_widget
+    end
+    if self.day then
+        day_widget = NumberPickerWidget:new{
+            show_parent = self,
+            value = self.day,
+            value_min = self.day_min or 1,
+            value_max = self.day_max or 31,
+            value_step = 1,
+            value_hold_step = self.day_hold_step or 3,
+            width = number_picker_widgets_width,
+        }
+        self:mergeLayoutInHorizontal(day_widget)
+    else
+        day_widget = dummy_widget
+    end
+
+    if self.hour then
+        hour_widget = NumberPickerWidget:new{
+            show_parent = self,
+            value = self.hour,
+            value_min = self.hour_min or 0,
+            value_max = self.hour_max or 23,
+            value_step = 1,
+            value_hold_step = self.hour_hold_step or 4,
+            width = number_picker_widgets_width,
+        }
+        self:mergeLayoutInHorizontal(hour_widget)
+    else
+        hour_widget = dummy_widget
+    end
+    if self.min then
+        min_widget = NumberPickerWidget:new{
+            show_parent = self,
+            value = self.min,
+            value_min = self.min_min or 0,
+            value_max = self.min_max or 59,
+            value_step = 1,
+            value_hold_step = self.min_hold_step or 10,
+            width = number_picker_widgets_width,
+        }
+        self:mergeLayoutInHorizontal(min_widget)
+    else
+        min_widget = dummy_widget
+    end
+    if self.sec then
+        sec_widget = NumberPickerWidget:new{
+            show_parent = self,
+            value = self.sec,
+            value_min = self.sec_min or 0,
+            value_max = self.sec_max or 59,
+            value_step = 1,
+            value_hold_step = self.sec_hold_step or 10,
+            width = number_picker_widgets_width,
+        }
+        self:mergeLayoutInHorizontal(sec_widget)
+    else
+        sec_widget = dummy_widget
+    end
+
+    separator_date = TextWidget:new{
+        text = "–",
+        face = self.title_face,
+        bold = true,
     }
-    self:mergeLayoutInHorizontal(month_hour_widget)
-    day_min_widget = NumberPickerWidget:new{
-        show_parent = self,
-        value = self.is_date and self.day or self.min,
-        value_min = self.min_min or self.day_min or (self.is_date and 1 or 0),
-        value_max = self.min_max or self.day_max or (self.is_date and 31 or 59),
-        value_step = 1,
-        value_hold_step = self.day_hold_step or self.min_hold_step or (self.is_date and 5 or 10),
-        date_month_hour = month_hour_widget,
-        date_year = year_widget,
+    separator_time = TextWidget:new{
+        text = _(":"),
```

Comment:
I don't think it makes sense to bother translators with this.

Reply:
Yep, I will clean that

Reply:
Thanks :)